specs/UserActivityHistory/UserActivityHistoryAPI.md

```diff
@@ -0,0 +1,243 @@
+
+## UserActivityHistoryQuery.ContentType property
+
+This property is a string that specifies the content type associated with the activity you want
+to retrieve. It allows the inclusion of an asterisk as a wildcard - e.g., "image/*" will match
+all content types beginning with "image/", such as "image/png", "image/jpeg", etc.
+This property is case-insensitive.
+
+## UserActivityHistoryQuery.EarliestStartTime property
+
+This is a nullable DateTime property that specifies the earliest start time of the activity
+you want to retrieve. Any activities with a StartTime property earlier than this will be excluded.
+If this property is left as null, it will be ignored.
+
+## UserActivityHistoryQuery.EarliestEndTime property
+
+This is a nullable DateTime property that specifies the earliest end time of the activity
+you want to retrieve. Any activities with an EndTime property earlier than this will be excluded.
+If this property is left as null, it will be ignored.
+
+## UserActivityHistoryQuery.LatestStartTime property
+
+This is a nullable DateTime property that specifies the latest start time of the activity
+you want to retrieve. Any activities with a StartTime property later than this will be excluded.
+If this property is left as null, it will be ignored.
+
+## UserActivityHistoryQuery.LatestEndTime property
+
+This is a nullable DateTime property that specifies the latest end time of the activity
+you want to retrieve. Any activities with an EndTime property later than this will be excluded.
+If this property is left as null, it will be ignored.
+
+## UserActivityHistoryOrderBy enum
+
+This enum specifies what property the results should be ordered by.  The options are as follows:
+
+| Name | Description |
+|-|-|
+| StartTime | Results will be in descending order of their StartTime property |
+| EndTime | Results will be in descending order of their EndTime property |
+| DwellTime | Results will be in descending order of the difference between their EndTime and StartTime properties |
+
+# API Details
+
+```c# (but really MIDL3)
+namespace Microsoft.Windows.ApplicationModel.UserActivities
+{
+    runtimeclass UserActivityHistory
+    {
+        static IVector<UserActivityHistoryItem> Search(
+            UserActivityHistoryQuery[] queries,
+            UserActivityHistoryOrderBy orderBy,
+            UInt32 maxResults); 
+
+        static IAsyncOperation<IVector<UserActivityHistoryItem> > SearchAsync( 
+            UserActivityHistoryQuery[] queries, 
+            UserActivityHistoryOrderBy orderBy, 
+            UInt32 maxResults);
+
+        static IVector<String> GetAppsWithUserActivity();
+
+        static IAsyncOperation<IVector<String>> GetAppsWithUserActivityAsync();
+    }
+
+    runtimeclass UserActivityHistoryItem
+    {
+        String AppName { get; };
+        String ActivityId { get; };
+        String DisplayText { get; };
+        String ContentType { get; };
+        String ContentUri { get; };
```

Comment:
what can the app do with this though? in general this won't be useful to apps other than the originator of this content, unless it's public, due to needing to authenticate.

Reply:
As long as the resource is something that can be accessed via a URI, e.g. a webpage, a document, etc., that URI can be used to bring it up to the user, which is the primary idea behind this API.